tika-parsers/src/main/java/org/apache/tika/parser/tagRatio/TextToTagRatio.java

```diff
@@ -0,0 +1,209 @@
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+
+/**
+ * Utility class for implementing TagRatio Parser. It has methods to compute the tag ratio
+ * and to read the xhtml document to do the computation.
+ *
+ * @since Apache Tika 1.13
+ */
+
+public class TextToTagRatio {
+    
+    private double[] linkTagList;
+    
+    
+    /**
+     * Calculate Text to Tag Ratio for raw html string
+     * @param Raw html passed as string
+     * @return list of tag ratios
+     */
+    
+    @SuppressWarnings("unchecked")
+    public double[] getTagRatioOfHtml(String html) {
+        
+        html = html.replaceAll("(?s)<!--.*?-->", "");
+        html = html.replaceAll("(?s)<script.*?>.*?</script>", "");
+        html = html.replaceAll("(?s)<SCRIPT.*?>.*?</SCRIPT>", "");
+        html = html.replaceAll("(?s)<style.*?>.*?</style>", "");
+        
+        try{
+            Parser p = new Parser(html);
+            NodeList nl = p.parse(null);
+            
+            BufferedReader br = new BufferedReader(new StringReader(nl.toHtml()));
+            int numLines = 0;
+            while (br.readLine() != null) {
+                numLines++;
+            }
+            br.close();
+            
+            linkTagList = new double[numLines];
+            HashMap<String, String> metaTagsMap = new HashMap<String, String>();
+            String line;
+            double threshold = 10;
+            StringBuffer sb = new StringBuffer();
+            double tagRatio = 0.0;
+            int count = 0;
+            br = new BufferedReader(new StringReader(nl.toHtml()));
+            for (int i = 0; i < linkTagList.length; i++) {
+                line = br.readLine();
+                line = line.trim();
+                if (line.equals("")) {
+                    continue;
+                }
+                linkTagList[i] = computeTextToTagRatio(line);
+                //Extract meta tags
+                populateMetaTags(line, metaTagsMap);
+                
+                if(linkTagList[i] != 0 && linkTagList[i] >= threshold){
+                    
+                    sb.append(line);
+                    sb.append("\n");
+                    tagRatio += linkTagList[i];
+                    count++;
+                }
+            }
+            
+            //Create a JSON Object
+            JSONObject obj = new JSONObject();
+            obj.put("avgTagRatio", (tagRatio/count));
+            obj.put("content", sb.toString());
+            
+            JSONArray array = new JSONArray();
+            array.put(metaTagsMap);
+            obj.put("meta-tags", array);
+            
+            
+            br.close();
+            
+        }catch (MalformedURLException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (ParserException e) {
+            e.printStackTrace();
+            return linkTagList;
+        }
```

Comment:
readability can be improved by making compact code, Checkout [multi-catch](http://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html).